### ISSUE TITLE
[Support]: Max CPU usage seems limited to 92-93%? Is it a Home assistant issue?

### ISSUE THREAD

**Issue by LaurenceGough** [blakeblackshear/frigate#3069]
### Describe the problem you are having

Hello,

I know I am not using a supported setup (no Coral TPU just yet) but I was just wondering if anyone could help me determine why my Frigate Add-On on my Home Assistant Supervised install (Installed on Raspberry Pi OS Lite) doesn't appear to be maxing out available CPU resources. Is it because Home Assistant is reserving some for itself? I have checked the last week or two and absolute peak is 93%, but generally leaving about 10% left which could be used for faster Inference speeds until a Coral arrives!

![image](https://user-images.githubusercontent.com/82294976/162545681-510decd3-eee2-487d-9615-314944f897cd.png)

Or perhaps it is just an inaccurate statistic?
CPU temps are nearly always below 40C with a large heatsink installed. I don't believe it's throttling. No power issues noted in Home Assistant OS under power status.
![image](https://user-images.githubusercontent.com/82294976/162545841-917cfe35-6022-4c2b-9d56-99f5ae863760.png)

### Version

0.11.0-CBF26E0

### Frigate config file

```yaml
detectors:
  cpu1:
    type: cpu
    num_threads: 4
  cpu2:
    type: cpu
    num_threads: 4

(I have tried all different combos and thread counts)
```

### Relevant log output

```shell
N/A
```

### FFprobe output from your camera

```shell
N/A
```

### Operating system

Other Linux

### Install method

HassOS Addon

### Coral version

CPU (no coral)

### Network connection

Wired

### Camera make and model

N/A

**Comment by NickM-27** [comment]
It doesn't exactly work like that, it's not like if a CPU used more resources it could have FASTER inference speeds. A lot that goes into inference speeds is IPC (instructions per clock), memory latency between instructions, memory latency with the video frames, etc. 

This means that using more CPU clocks won't help it, it is accomplishing the task just fine but the hardware only allows it to work so fast. The reason the coral is so much faster is because it has hardware built to run these calculations specifically so it takes less instructions and can run those specific instructions much faster than a general purpose CPU.

**Comment by LaurenceGough** [comment]
> It doesn't exactly work like that, it's not like if a CPU used more resources it could have FASTER inference speeds (assuming a reasonable starting point). A lot that goes into inference speeds is IPC (instructions per clock), memory latency between instructions, memory latency with the video frames, etc.
> 
> This means that using more CPU clocks won't help it, it is accomplishing the task just fine but the hardware only allows it to work so quickly. The reason the coral is so much faster is because it has hardware built to run these calculations specifically so it takes less instructions and can run those specific instructions much faster than a general purpose CPU.

Thanks Nick, that is a very fair point. I understand about the TPU being designed for this task but I was just wondering if there was any reason why my CPU wasn't being utilised fully, perhaps something as simple as home assistant reserving some of the CPU so that it can function even if an addon was to use a lot of resources, you never know it might help a little bit. And Ugghhh, my Coral TPU order has just been pushed back another month this morning :(

**Comment by NickM-27** [comment]
> Thanks Nick, that is a very fair point. I understand about the TPU being designed for this task but I was just wondering if there was any reason why my CPU wasn't being utilised fully, perhaps something as simple as home assistant reserving some of the CPU so that it can function even if an addon was to use a lot of resources, you never know it might help a little bit.

Back before I had my coral, my server only used ~50% CPU when running detection and it was still ~100ms inference speed, this is just due to the fact that there's other factors for inference speed other than how fast the CPU cores are running. I suppose it is possible the OS is reserving some resources, but I have seen Frigate hit 100% for other users on HA OS.

> And Ugghhh, my Coral TPU order has just been pushed back another month this morning :(

That stinks

**Comment by LaurenceGough** [comment]
I have since read/learnt that Frigate AI object detection is only designed to work on one CPU at a time (or thread, I cannot recall now) I believe as of course this project is meant to be used with a Coral TPU. It does state for testing only use CPU.

Sad luck for me though as availability of a Coral TPU pushed back even further.

I also found for some reason that my Raspberry Pi 4 was not boosting due to installing HomeAssistantOS on it previously which sets a boot config file that doesn't allow for boost (1.4GHz only).
I've now overclocked it to 2GHz on all cores which has helped speed up AI detection massively so I hope this helps someone reading (make sure you have decent cooling).

Thanks.
